File: api/src/main/java/com/ruby/rubia_server/core/dto/CreateUserDTO.java
package com.ruby.rubia_server.core.dto;

import com.ruby.rubia_server.core.enums.UserRole;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateUserDTO {
    
    @NotBlank(message = "Nome é obrigatório")
    @Size(min = 2, max = 255, message = "Nome deve ter entre 2 e 255 caracteres")
    private String name;
    
    @NotBlank(message = "Email é obrigatório")
    @Email(message = "Email deve ter formato válido")
    private String email;
    
    @NotBlank(message = "Senha é obrigatória")
    @Size(min = 6, max = 100, message = "Senha deve ter entre 6 e 100 caracteres")
    private String password;
    
    @NotNull(message = "ID da empresa é obrigatório")
    private UUID companyId;
    
    @NotNull(message = "ID do departamento é obrigatório")
    private UUID departmentId;
    
    @NotNull(message = "Role é obrigatório")
    private UserRole role;
    
    private String avatarUrl;
    
    private LocalDate birthDate;
    
    private Double weight;
    
    private Double height;
    
    @Size(max = 500, message = "Endereço deve ter no máximo 500 caracteres")
    private String address;
}

File: api-dotnet/Rubia.Server/DTOs/CreateUserDto.cs
using System.ComponentModel.DataAnnotations;
using Rubia.Server.Enums;

namespace Rubia.Server.DTOs;

public class CreateUserDto
{
    [Required(ErrorMessage = "Nome é obrigatório")]
    [StringLength(255, MinimumLength = 2, ErrorMessage = "Nome deve ter entre 2 e 255 caracteres")]
    public string Name { get; set; } = string.Empty;

    [Required(ErrorMessage = "Email é obrigatório")]
    [EmailAddress(ErrorMessage = "Email deve ter formato válido")]
    public string Email { get; set; } = string.Empty;

    [Required(ErrorMessage = "Senha é obrigatória")]
    [StringLength(100, MinimumLength = 6, ErrorMessage = "Senha deve ter entre 6 e 100 caracteres")]
    public string Password { get; set; } = string.Empty;

    [Required(ErrorMessage = "ID da empresa é obrigatório")]
    public Guid CompanyId { get; set; }

    [Required(ErrorMessage = "ID do departamento é obrigatório")]
    public Guid DepartmentId { get; set; }

    [Required(ErrorMessage = "Role é obrigatório")]
    public UserRole Role { get; set; }

    public string? AvatarUrl { get; set; }

    public DateOnly? BirthDate { get; set; }

    public double? Weight { get; set; }

    public double? Height { get; set; }

    [StringLength(500, ErrorMessage = "Endereço deve ter no máximo 500 caracteres")]
    public string? Address { get; set; }
}